packages/meditrak-server/src/kobo/startSyncWithKoBo.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import keyBy from 'lodash.keyby';
+import winston from 'winston';
+import { DataBroker } from '@tupaia/data-broker';
+import { generateId } from '@tupaia/database';
+import { reformatDateStringWithoutTz } from '@tupaia/utils';
+
+const PERIOD_BETWEEN_SYNCS = 10 * 60 * 1000; // 10 minutes between syncs
+const SERVICE_NAME = 'KoBo';
+
+export async function startSyncWithKoBo(models) {
+  if (process.env.KOBO_SYNC_DISABLE) {
+    // eslint-disable-next-line no-console
+    console.log('KoBo sync is disabled');
+  } else {
+    const dataBroker = new DataBroker();
+    setInterval(() => syncWithKoBo(models, dataBroker), PERIOD_BETWEEN_SYNCS);
+  }
+}
+
+export async function syncWithKoBo(models, dataBroker) {
+  await pullLatest(models, dataBroker);
+}
+
+async function pullLatest(models, dataBroker) {
+  const syncCursor = await models.syncCursor.findOne({ service_name: SERVICE_NAME });
+
+  // Pull data from KoBo
+  const koboData = await dataBroker.pull(
+    {
+      code: syncCursor.config.koboSurveys,
+      type: dataBroker.getDataSourceTypes().SYNC_GROUP,
+    },
+    {
+      startSubmissionTime: reformatDateStringWithoutTz(syncCursor.sync_time),
+    },
+  );
+
+  const apiUser = await models.apiClient.findOne({ username: process.env.KOBO_API_USERNAME });
+  if (!apiUser) {
+    throw new Error('Cannot find API client user for KoBo');
+  }
+
+  // Create new survey_responses
+  let newSyncTime = syncCursor.sync_time;
+  await models.wrapInTransaction(async transactingModels => {
+    for (const [surveyCode, responses] of Object.entries(koboData)) {
+      const survey = await transactingModels.survey.findOne({ code: surveyCode });
+
+      for (const responseData of responses) {
+        if (responseData.eventDate > newSyncTime) {
+          newSyncTime = responseData.eventDate;
+        }
+        const entity = await transactingModels.entity.findOne({ code: responseData.orgUnit });
+        if (!entity) {
+          winston.warn(`Skipping KoBo sync for unknown entity ${responseData.orgUnit}`);
+          continue;
+        }
+
+        const surveyResponse = await transactingModels.surveyResponse.create({
+          id: generateId(),
+          survey_id: survey.id,
+          user_id: apiUser.user_account_id,
+          assessor_name: 'KoBo Integration',
```

Comment:
Although we are pretty limited to using a generic user id, we can probably do a better job here - the KoBo data payload includes "username", which is not found in the ones I've done on the test account but might end up defined. Can we use that instead? I realise it means breaking the strict event format, but that's ok with me because it's in an additive way so future consumers of the data as an actual event would still get everything they needed (and more)